sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AADOAuth2AuthenticatedPrincipal.java

```diff
@@ -32,18 +33,28 @@
 
     private final String tokenValue;
 
+    private final String name;
+
     private JWTClaimsSet jwtClaimsSet;
 
     public AADOAuth2AuthenticatedPrincipal(Map<String, Object> headers,
                                            Map<String, Object> attributes,
                                            Collection<GrantedAuthority> authorities,
                                            String tokenValue) {
+        this(headers, attributes, authorities, tokenValue, null);
+    }
+
+    public AADOAuth2AuthenticatedPrincipal(Map<String, Object> headers,
+                                           Map<String, Object> attributes,
+                                           Collection<GrantedAuthority> authorities,
+                                           String tokenValue, String name) {
         Assert.notEmpty(attributes, "attributes cannot be empty");
         Assert.notEmpty(headers, "headers cannot be empty");
         this.headers = headers;
         this.tokenValue = tokenValue;
         this.attributes = Collections.unmodifiableMap(attributes);
         this.authorities = authorities == null ? NO_AUTHORITIES : Collections.unmodifiableCollection(authorities);
+        this.name = (name != null) ? name : (String) this.attributes.get("sub");
```

Comment:
So the default claim changed from `name` to `sub`?

And I create a PR to simplify code: https://github.com/ZhuXiaoBing-cn/azure-sdk-for-java/pull/1

Reply:
Yes, the default is to get the `sub` claim.